### ISSUE TITLE
Feature: Render solution default to display resolution / native

### ISSUE THREAD

**Issue by TCB13** [X-Hax/SA-Mod-Manager#232]
Hi,

![image](https://github.com/user-attachments/assets/26c77933-91f3-46f0-b79b-f0e3745231bf)

It would be great if by default / on first launch SA Mod Manager automatically set the render resolution to be the same as the default display resolution.

Right now If I go into `Default.json` and delete this:

```
    "HorizontalResolution": 2560,
    "VerticalResolution": 1440,
```

It will default to something like 640px. Why not default to the main screen resolution instead?

Thank you.

**Comment by PiKeyAr** [comment]
640x480+windowed mode is the failsafe combination. For first startup, see #46.

**Comment by TCB13** [comment]
> 640x480+windowed mode is the failsafe combination

What do you mean by failsafe? Isn't it enough to make sure the game renders at the maximum resolution supported by the system? Are there any cases where it may fail?

Thanks.

**Comment by PiKeyAr** [comment]
We've had issues in the past with the game failing to set the correct window size and crashing silently, especially in the borderless fullscreen mode. 640x480+windowed is the combination that should work in the majority of scenarios, so I'd argue for that being the default. Anyway it's just the default value that is used when the actual value is missing in the JSON file.

I do think it would be good if the Manager could apply the optimal settings, including the rendering resolution, on first run.

**Comment by TCB13** [comment]
> I do think it would be good if the Manager could apply the optimal settings, including the rendering resolution, on first run.

Yeah, me too, the question is, what's the difference between applying "apply the optimal settings, including the rendering resolution" on first run vs doing the same thing when the value is missing from the JSON?

I was assuming the way you would implement the first run settings would be the same as when it doesn't find a property in the JSON file.

**Comment by ItsEasyActually** [comment]
I went in and implemented this. To keep it as unintrusive as possible, a native resolution will be set on initial game setup (ie when booting the mod manager for the first time and it finds a game or selecting one for the first time in the drop down) if there is no profile migration on initial setup.

Completed in [4ff074b](https://github.com/X-Hax/SA-Mod-Manager/commit/4ff074b9538faace16fede7462792fc7e6c98980).

As for why it has to be done this way, it's because you can't really make an arbitrary default value for a configuration setup like that. Rather, it's probable, but it's not necessary and risky to setup imo. Much better to have a safe fallback.

**Comment by TCB13** [comment]
> I went in and implemented this. To keep it as unintrusive as possible, a native resolution will be set on initial game setup (ie when booting the mod manager for the first time and it finds a game or selecting one for the first time in the drop down) if there is no profile migration on initial setup.
> 
> Completed in [4ff074b](https://github.com/X-Hax/SA-Mod-Manager/commit/4ff074b9538faace16fede7462792fc7e6c98980).
> 
> As for why it has to be done this way, it's because you can't really make an arbitrary default value for a configuration setup like that. Rather, it's probable, but it's not necessary and risky to setup imo. Much better to have a safe fallback.

Woow! That's great, thank.

**Comment by PiKeyAr** [comment]
Might be a good idea to also set window mode to Borderless when it does that, otherwise it will be trying to create a window larger than screen size.

**Comment by ItsEasyActually** [comment]
I went in and did just that in [28a32d6](https://github.com/X-Hax/SA-Mod-Manager/commit/28a32d6b4c9fb14258bceabb6baf37e4b82c1c9c). Also added a check to ensure the data is valid in the event it fails to get the screen bounds for some reason.
